src/runtime/memory/memory_manager.cc

```diff
@@ -101,6 +100,11 @@ NDArray StorageObj::AllocNDArray(size_t offset, ShapeTuple shape, DLDataType dty
   // buffer intact.
   container->manager_ctx = reinterpret_cast<void*>(this);
 
+  // The only change we make w.r.t offset is modifying the data pointer
+  // of the backing tensor to point into the buffer instead of its start.
+  auto offset_ptr = reinterpret_cast<uint8_t*>(this->buffer.data) + offset;
```

Comment:
This change has been submitted to main: https://github.com/apache/tvm/pull/15912